infra/modules/azure_postgres_server/variables.tf

```diff
@@ -0,0 +1,271 @@
+#---------#
+# General #
+#---------#
+
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "environment" {
+  type = object({
+    prefix          = string
+    env_short       = string
+    location        = string
+    domain          = optional(string)
+    app_name        = string
+    instance_number = string
+  })
+
+  validation {
+    condition     = length("${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : replace(var.environment.domain, "-", "")}${var.environment.app_name}-psql-replica-${var.environment.instance_number}") <= 63
+    error_message = "Azure PostgreSQL Flexible Server name must contain between 3 and 63 characters. Current value is \"${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : var.environment.domain}${var.environment.app_name}-psql-replica-${var.environment.instance_number}\""
+  }
+
+  description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "db_version" {
+  type        = number
+  description = "The version of PostgreSQL Flexible Server to use. Possible values are 11, 12, 13, 14, 15 and 16"
+  default     = 16
+}
+
+#------------#
+# Networking #
+#------------#
+
+variable "private_dns_zone_id" {
+  type        = string
+  description = "ID of the private DNS zone"
+}
+
+variable "private_dns_zone_resource_group_name" {
```

Comment:
ditto

Reply:
This is used in `data.tf`, we have discuss about it in this [conversation](https://github.com/pagopa/dx/pull/101/files/609547980b61934102202de8b6a694ca2f273164#diff-0f780a63ee126c9c582184b0e7a1a9bd2390cb92789eeaf03ec04a6e3c46e2d2)